torch_xla/csrc/ir.h

```diff
@@ -132,9 +138,13 @@ class XlaNode : public torch::lazy::Node {
 
   static std::vector<torch::lazy::SourceLocation> GetFrameInfo();
 
+  static torch::lazy::hash_t CreateShardingHash(
+      std::shared_ptr<xla::OpSharding> sharding, torch::lazy::hash_t hash_seed);
+
   xla::Shape xla_shape_;
   torch::lazy::hash_t node_hash_ = 0;
   torch::lazy::hash_t dag_hash_;
+  torch::lazy::hash_t sharding_hash_ = 0;
```

Comment:
How does sharding_hash_ get used, is it combined to node_hash_ to represent the final node hash?

Reply:
In the subsequent PR (https://github.com/pytorch/xla/pull/4287), we combine it with the dag hash when `hash()` is called.

Reply:
I think we should just combined it with `dag_hash_` during construction and not maintaining it as a separate state.

Reply:
In subsequent PRs I combine it with dag hash and then move that logic into the constructor, this was just meant to be an intermediate state.